R/Connection.R

```diff
@@ -69,6 +69,98 @@ setClass(
+#' @name odbcConnectionColumns
+#'
+#' @description For a given table this function recturns detailed information on
+#' all fields / columns.  The expectation is that this is a relatively thin
+#' wrapper around the ODBC SQLColumns function call.
+#'
+#' @details For more information see SQLColumns ODBC function
+#' @param conn OdbcConnection
+#' @param name table we wish to get information on
+#' @param ... additional parameters to methods
+#'
+#' @seealso The ODBC documentation on [SQLColumns](https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlcolumns-function)
+#' for further details on the expected output.
+#'
+#' @return data.frame with columns
+#' \itemize{
+#'   \item{table_cat}
+#'   \item{table_schem}
+#'   \item{table_name}
+#'   \item{column_name}
+#'   \item{data_type}
+#'   \item{type_name}
+#'   \item{column_size}
+#'   \item{buffer_length}
+#'   \item{decimal_digits}
+#'   \item{numeric_precision_radix}
+#'   \item{nullable}
+#"   \item{remarks}
+#'   \item{column_def}
+#'   \item{sql_data_type}
+#'   \item{sql_datetime_subtype}
+#'   \item{char_octet_length}
+#'   \item{ordinal_position}
+#' }
+#' @export
+setGeneric(
+  "odbcConnectionColumns",
+  valueClass = "data.frame",
+  function(conn, name, ...) {
+    standardGeneric("odbcConnectionColumns")
+  }
+)
+
+#' @name odbcConnectionColumns
+#' @aliases odbcConnectionColumns,OdbcConnection,Id-method
+#' @inheritParams odbcConnectionColumns
+#' @param column_name The name of the column to return, the default returns all columns.
+#'
+#' @export
+setMethod(
+  "odbcConnectionColumns",
+  c("OdbcConnection", "Id"),
+  function(conn, name, column_name = NULL) {
+
+    odbcConnectionColumns(conn,
+      name = name@name[["table"]],
+      catalog_name = name@name[["catalog"]],
+      schema_name = name@name[["schema"]],
+      column_name = column_name)
+  }
+)
+
+#' @name odbcConnectionColumns
+#' @aliases odbcConnectionColumns,OdbcConnection,character-method
+#' @inheritParams odbcConnectionColumns
+#' @param catalog_name charaacter catalog where the table is located
+#' @param schema_name charaacter schema where the table is located
+#' @export
+setMethod(
+  "odbcConnectionColumns",
+  c("OdbcConnection", "character"),
+  function(conn, name, catalog_name = NULL, schema_name = NULL, column_name = NULL) {
+    detail <-
+      connection_sql_columns(conn@ptr,
+        table_name = name,
+        catalog_name = catalog_name,
+        schema_name = schema_name,
+        column_name = column_name)
+    # Re-order/rename columns to make output ODBC-like
+    detail <- detail[, c("catalog_name", "schema_name", "table_name", "name",
+      "data_type", "field.type", "column_size", "buffer_length",
+      "decimal_digits", "numeric_precision_radix", "nullable", "remarks",
+      "column_default", "sql_data_type", "sql_datetime_subtype",
+      "char_octet_length", "ordinal_position")]
+    names(detail)[c(1, 2, 4, 6, 13)] <- c("table_cat", "table_schem",
```

Comment:
I think I would prefer to _not_ rename. And do we need to reorder the columns?

Reply:
Jim my original thinking here was to make the output as close to the output of the ODBC `SQLColumns` function as possible.   It seemed to me that, since we are offering the end-user the ability to write their own implementation, then it is more natural to anchor on the field naming of the ODBC's `SQLColumns` API, than on the choices made when implementing the internal `connection_sql_columns`.

If you also think that's worthwhile then we either re-name the columns here, or change the output of `connection_sql_columns`.  The latter seemed less desirable since even though internal, I wasn not sure if there aren't folks that have developed dependencies.

The re-ordering was more thinking in the same direction - making the output align with what I can see when I execute `sp_columns` in my SQL Server client for example.

I don't feel strongly here - happy to make this just a wrapper around `connection_sql_columns` if you think that's the best way forward.

Reply:
I would prefer it to just be a wrapper around `connection_sql_columns`